### DISCUSSION TITLE
22.12

### DISCUSSION THREAD

**Discussion by falkTX** [DISTRHO/Cardinal#461] [Announcements]
General changes/fixes
- Add portable standalone Windows binaries
- Add JTB Waves example/demo patch
- Fix SSE not being enabled in pffft for 32bit builds
- Improve docs for building Windows binaries
- Stop using LTO for macos-universal builds

Module updates
- Add Sapphire
- Add Stoermelder Pack-one
- Add Surge-XT
- Update BaconPlugs
- Update Befaco
- Update MindMeld
- Update PathSet

Web-assembly related changes
- Add Web-assembly build instructions
- Enable brotli compression on release files
- Enable emscripten --use-preload-cache flag, improves browser cache
- Fix keyboard input having wrong keys
- Fix mouse lock on chrome-based browsers
- Fix versioning

Special Notes:
- Building now requires a compiler compatible with C++17
- Windows builds now require 0x0600/Vista, needed for Surge

<hr /><em>This discussion was created from the release <a href='https://github.com/DISTRHO/Cardinal/releases/tag/22.12'>22.12</a>.</em>

**Comment by ThePython10110** [comment]
PackOne and Surge! YES!

**Comment by jeffsnail12** [comment]
Hi . Someone Can tell me what PC config to do run Cardinal ? My old AMD FX not enough !

**Reply by falkTX** [reply]
Anything that can do SSE3.
Windows  >= Vista, macOS >= 10.8 or any recent Linux distribution.

**Reply by jeffsnail12** [reply]
my Fx is sse3 compatible, but CPU overload quickly ! . I wonder which CPU to choose as a replacement. i7, i9, ryzen? Is the benchmark test reliable? no Mac, too expensive for me.

**Reply by dromer** [reply]
I could use Cardinal on a 1st gen i5 (ok not the biggest/complex patches, but enough to play with it) so anything newer should be fine.

**Reply by subderground** [reply]
Hi, Perhaps try to set a higher Asio buffer size on your soundcard (latence)?

ps: Merry Xmass & Happy 23. Thanks again for this great Cardinal plug-in 🎅🤗

**Reply by jeffsnail12** [reply]
Hi . My set  is 256 samples . Latence is to max acceptable level , but i can do intersting patch all the same . Available on patchstorage .
I 'll change my PC later . Merry xmax , have a good year 2023 😁

**Comment by Hipwell** [comment]
Sorry, I'm totally new to this. Keen to run this on a Mac mini with Pro Tools but no idea what files from the list provided here I need to download to get it going? Again, apologies for the rookie question and much gratitude for the help! :)

**Reply by dromer** [reply]
MacOS Universal has it in the name: should work on all Macs, both Intel and M1.

**Reply by Hipwell** [reply]
Thank you, thank you!

**Comment by ThePython10110** [comment]
I don't own a Mac myself, so my advice could be wrong, but it looks like you should probably download 
[Cardinal-macOS-intel-22.12.pkg](https://github.com/DISTRHO/Cardinal/releases/download/22.12/Cardinal-macOS-intel-22.12.pkg)
 or [Cardinal-macOS-universal-22.12.pkg](https://github.com/DISTRHO/Cardinal/releases/download/22.12/Cardinal-macOS-universal-22.12.pkg), depending on whether you have an Intel Mac Mini or not. To find out, click the Apple button, then About This Mac, then look at the Chip.

**Reply by Hipwell** [reply]
Legend, thanks so much! :)

**Comment by UMCorps** [comment]
Can't get Cardinal AU to work in sandboxed AU hosts on Mac OS.

Its a long-standing issue. Was first identified in this thread on the Juice forum

https://forum.juce.com/t/sandboxsafe-and-macos-host-au2-hardened-runtime-disable-library-validation/47281



Was also an issue with Surge XT which got discussed and resolved in this thread

https://github.com/surge-synthesizer/surge/issues/5633



Any chance of a similar resolution for Cardinal?

**Comment by GBaige** [comment]
Could we please have an optional setting to prevent Cardinal from creating Surge XT directories under Documents in the next release? They should only be created if the user wants to load a custom wave table, preset, etc., not every time any Surge XT module is loaded. If anything, Cardinal should look into its own resource directories rather than creating directories elsewhere and expecting files there. This isn't self contained.

**Reply by cosinekitty** [reply]
This is really a Surge XT issue, not a Cardinal issue. @baconpaul might want to respond to this? Maybe there is a way to configure Surge XT to locate the directory elsewhere or simply not create it?

**Reply by baconpaul** [reply]
It’s fixed in surge xt 2.1.0

**Reply by baconpaul** [reply]
And 2.1.1 also which is now in the library queue

**Reply by GBaige** [reply]
Great, thanks. Now I wonder how long until the next Cardinal update or whether it will include updated Surge...

**Reply by dromer** [reply]
Whenever a volunteer gets around to do it. I had a quick look and it will require at least a little bit of work (had some build errors due to changes and wasn't directly obvious what. SurgeXT is quite non-trivial to build compared to other modules).

I do think/hope a 23.02 release is likely since there are some other updated modules as well.

**Reply by baconpaul** [reply]
we changed to the alternative cmake based build system at [qno/vcv-plugin-cmake-example](https://github.com/qno/vcv-plugin-cmake-example) for surge, bacon plugs, and the new air win modules. Might be work for cardinal to support that. I know you build in lots of odd environments and stuff.

If you want, you can just take the 2.0.3 makefile, add in the new modules and check it in as Makefile.cardinal and upstream it if that makes your life easier.
